lib/libutee/tee_api_arith_mpi.c

```diff
@@ -443,7 +443,23 @@ void TEE_BigIntNeg(TEE_BigInt *dest, const TEE_BigInt *src)
 void TEE_BigIntMul(TEE_BigInt *dest, const TEE_BigInt *op1,
 		   const TEE_BigInt *op2)
 {
-	bigint_binary(dest, op1, op2, mbedtls_mpi_mul_mpi);
+	size_t bs1 = TEE_BigIntGetBitCount(op1);
+	size_t bs2 = TEE_BigIntGetBitCount(op2);
+	size_t s = TEE_BigIntSizeInU32(bs1) + TEE_BigIntSizeInU32(bs2);
+	TEE_BigInt zero[TEE_BigIntSizeInU32(1)];
+	TEE_BigInt *tmp = NULL;
+
+	tmp = (TEE_BigInt *)TEE_Malloc(4 * s, 0);
```

Comment:
We should allocate using `mempool_alloc()` instead. We still need to check that the allocation succeeds, but it will never fail due to exhaustion of the heap. Also, please replace `4` with `sizeof(uint32_t)`.

Reply:
OK